Allow taxons to be deleted

Davidslv opened the pull request:
Trello: https://trello.com/c/vBJSZakA/93-allow-taxons-to-be-deleted

Flow:

When the user presses the `Delete` button, and the taxon is a Parent taxon, the children of that taxon no longer have that taxon as parent. Also documents with such taxon no longer will have the taxon tagged to them.

This is all possible by requesting Publishing API to unpublish a given taxon, with the type "gone"

### Look & Feel

<img width="1401" alt="screen shot 2016-08-16 at 15 46 09" src="https://cloud.githubusercontent.com/assets/136777/17703191/c16124d6-63c8-11e6-9d64-da40568bbae5.png">

mobaig commented on app/controllers/taxons_controller.rb:
```diff
@@ -48,6 +48,18 @@ def update
     redirect_to taxons_path
   end
 
+  def destroy
```
Is this already tested somewhere? If not, can we add one?

Davidslv replied:
I've added a controller spec for it now. Thanks

mobaig commented on app/controllers/taxons_controller.rb:
```diff
@@ -48,6 +48,18 @@ def update
   end
 
+  def destroy
+    resp = Services.publishing_api.unpublish(params[:id], type: "gone")
```
Rather we didn't have the abbreviated variable name here. Reckon you can you just be rid of it by renaming it to `response_code` and moving the `.code` call below to this line.

mobaig commented on app/controllers/taxons_controller.rb:
```diff
@@ -48,6 +48,18 @@ def update
+  def destroy
+    resp = Services.publishing_api.unpublish(params[:id], type: "gone")
+
+    message = if resp.code == 200
```
Pretty sure you'll get the linter complaining here about the alignment and indentation of the if block.

You can resolve this by either wrapping the whole thing in a begin/end, or by lining up the `else` and `end` with the `if`.  I think the latter is ugly as hell but either would do the job.

jackscotti commented:
Did you get input from a designer on the button colours? I have never seen the light blue being used on GOV.UK. @alextea

tijmenb commented on app/views/taxons/index.html.erb:
```diff
@@ -19,8 +18,18 @@
+          <%= link_to 'View', taxon_path(taxon['content_id']),
+            class: 'view-tagged-content btn btn-primary' %>
+
+          <%= link_to 'Edit', edit_taxon_path(taxon['content_id']),
```
"View" and "Edit" are links to pages, so they need to look like links ("Delete" is an action, so it's good that it's a button).

carvil commented on app/controllers/taxons_controller.rb:
```diff
@@ -48,6 +48,18 @@ def update
+    resp = Services.publishing_api.unpublish(params[:id], type: "gone")
+
+    message = if resp.code == 200
+      { success: 'You have successfully deleted the taxon' }
```
It would be nice to move those to I18n as well (both the success and the alert)

Davidslv commented:
@tijmenb, @jackscotti, @alextea  I've reverted the buttons on View & Edit, they remain as links
@carvil I've added `I18n` translations
@mobaig I've created a private method inside the controller instead

Thanks!

Davidslv commented:
Updated description.
